### PR TITLE
DOP-4021: Migrate gatsby plugin to docsets

### PR DESCRIPTION
**Pull request by mmeigs**
### Stories/Links:

DOP-4021

### Current Behavior:

[Current](https://www.mongodb.com/docs/atlas/)

### Staging Links:

[Staging](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/matt.meigs/DOP-4021-plugin-docsets/index.html)
Nothing should be different! :)

### Notes:

This PR ensures that the requests to the `repos_branches` collection in the Gatsby plugin(s) are now routed through the source of truth in the `docsets` collection. This mainly switches fetchRepoBranches to `fetchDocset`.

### CODE REVIEW

File: plugins/gatsby-source-snooty-prod/gatsby-node.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ const createRemoteMetadataNode = async ({ createNode, createNodeId, createConten
   const productList = manifestMetadata?.associated_products || [];
```
Comment (schmalliso) [inline plugins/gatsby-source-snooty-prod/gatsby-node.js:32]: We don't render associations in staging, so I'm not sure why this code exists anyway…

Comment (mmeigs) [reply plugins/gatsby-source-snooty-prod/gatsby-node.js:32]: Oh, this is in prod!

**Review by rayangler — APPROVED** [review]

File: plugins/gatsby-source-snooty-prod/gatsby-node.js (hunk 1 above)
Comment (schmalliso) [reply plugins/gatsby-source-snooty-prod/gatsby-node.js:32]: I shouldn't comment on PRs that early in the morning. Please ignore me :)

Comment (mmeigs) [reply plugins/gatsby-source-snooty-prod/gatsby-node.js:32]: I will never ignore you!!

**Pull request merged**